example/stochastic_subgrid/stochastic_subgrid_package.cpp

```diff
@@ -0,0 +1,462 @@
+//========================================================================================
+// (C) (or copyright) 2020-2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "stochastic_subgrid_package.hpp"
+
+#include <algorithm>
+#include <chrono> // NOLINT [build/c++11]
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <Kokkos_Random.hpp>
+
+#include <coordinates/coordinates.hpp>
+#include <parthenon/package.hpp>
+
+#include "kokkos_abstraction.hpp"
+#include "reconstruct/dc_inline.hpp"
+#include "utils/alias_method.hpp"
+
+using namespace parthenon::package::prelude;
+using namespace parthenon::AliasMethod;
+
+// *************************************************//
+// define the "physics" package Advect, which      *//
+// includes defining various functions that control*//
+// how parthenon functions and any tasks needed to *//
+// implement the "physics"                         *//
+// *************************************************//
+
+namespace stochastic_subgrid_package {
+
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
```

Comment:
This is a large function I would recommend breaking this up into smaller local functions in an unnames namespace. I think a lot of the variables in this function can be declared const, that would help clarify where things are being changed vs being used.

Reply:
I don't think breaking it up makes sense, because there is no code reuse here that could be extracted. So breaking up this function into smaller functions is the same as breaking it up into scopes with comment headers that delineate different sections. That's what I've done now. I think that's a better way to go because it doesn't unnecessarily bloat the code with functions that are called only once.

I've rearranged the reading and adding parameters to the package to keep things as local as possible and I've also added const to the variables that are not changed.

Reply:
Fair enough this is on the side of stylistic preference, you do gain one more advantage by placing things in functions, you can fit more into your screen.

Reply:
Actually, scopes are even better in this regard, I think. You don't have to waste lines pulling in variables you need from the outer scope and you can place the "function body" right where you need it (so no jumping around), and you can fold it just the same to make space for more code on the screen :)